src/main/java/org/galatea/starter/utils/DateChecker.java

```diff
@@ -2,84 +2,81 @@
 
 import java.time.DayOfWeek;
 import java.time.LocalDate;
-import java.util.Arrays;
-import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-public class DateChecker {
-  static HashSet<String> holidayDates;
-
-  public DateChecker() {
-    holidayDates = new HashSet<>(Arrays.asList(
-        "2021-01-01",
-        "2021-01-18",
-        "2021-02-15",
-        "2021-04-2",
-        "2021-05-31",
-        "2021-07-04",
-        "2021-09-06",
-        "2021-11-25",
-        "2021-12-24",
-        "2020-01-01",
-        "2020-01-20",
-        "2020-02-17",
-        "2020-04-10",
-        "2020-05-25",
-        "2020-07-04",
-        "2020-09-07",
-        "2020-11-26",
-        "2020-12-25",
-        "2019-01-01",
-        "2019-01-21",
-        "2019-02-18",
-        "2019-03-19",
-        "2019-04-27",
-        "2019-07-04",
-        "2019-09-02",
-        "2019-11-28",
-        "2019-12-25",
-        "2018-01-01",
-        "2018-01-15",
-        "2018-02-19",
-        "2018-03-30",
-        "2018-04-28",
-        "2018-07-04",
-        "2018-09-03",
-        "2018-11-22",
-        "2018-12-25",
-        "2017-01-02",
-        "2017-01-16",
-        "2017-02-20",
-        "2017-03-14",
-        "2017-04-29",
-        "2017-07-04",
-        "2017-09-04",
-        "2017-11-23",
-        "2017-12-25",
-        "2016-01-01",
-        "2016-01-18",
-        "2016-02-15",
-        "2016-03-25",
-        "2016-04-30",
-        "2016-07-04",
-        "2016-09-05",
-        "2016-11-24",
-        "2016-12-26",
-        "2015-01-01",
-        "2015-01-19",
-        "2015-02-16",
-        "2015-03-03",
-        "2015-04-25",
-        "2015-07-03",
-        "2015-09-07",
-        "2015-11-26",
-        "2015-12-25"));
-  }
+public final class DateChecker {
+  static Set<String> holidayDates = Stream.of("2021-01-01",
```

Comment:
You should not hardcode lists of dates like this. What are some better ways to determine this?

Reply:
I found a way of calculating the weekends, but couldn't figure out an efficient way of determining exchange holidays in a cleaner way - mostly because there was a lot of variation from year to year.